process/accountProcessor.go

```diff
@@ -132,6 +132,67 @@ func (ap *AccountProcessor) GetESDTTokenData(address string, key string) (*data.
 	return nil, ErrSendingRequest
 }
 
+// GetESDTsWithRole returns the token identifiers where the given address has the given role assigned
+func (ap *AccountProcessor) GetESDTsWithRole(address string, role string) (*data.GenericAPIResponse, error) {
+	observers, err := ap.proc.GetObservers(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		apiResponse := data.GenericAPIResponse{}
+		apiPath := AddressPath + address + "/esdts-with-role/" + role
+		respCode, err := ap.proc.CallGetRestEndPoint(observer.Address, apiPath, &apiResponse)
+		if err == nil || respCode == http.StatusBadRequest || respCode == http.StatusInternalServerError {
+			log.Info("account ESDTs with role",
+				"address", address,
+				"role", role,
+				"shard ID", observer.ShardId,
+				"observer", observer.Address,
+				"http code", respCode)
+			if apiResponse.Error != "" {
+				return nil, errors.New(apiResponse.Error)
+			}
+
+			return &apiResponse, nil
+		}
+
+		log.Error("account get ESDTs with role", "observer", observer.Address, "address", address, "role", role, "error", err.Error())
+	}
+
+	return nil, ErrSendingRequest
+}
+
+// GetOwnedNFTs returns the token identifiers of the NFTs where the given address is the owner
+func (ap *AccountProcessor) GetOwnedNFTs(address string) (*data.GenericAPIResponse, error) {
+	observers, err := ap.proc.GetObservers(core.MetachainShardId)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		apiResponse := data.GenericAPIResponse{}
+		apiPath := AddressPath + address + "/owned-nfts/"
+		respCode, err := ap.proc.CallGetRestEndPoint(observer.Address, apiPath, &apiResponse)
+		if err == nil || respCode == http.StatusBadRequest || respCode == http.StatusInternalServerError {
```

Comment:
a lot of duplicated code. The only difference is apiPath line between this and the previous function. You could have one common function where you send that apiPath. Plus apiPath can be initialized only once.

Reply:
I know what you mean, but it is a simple logic of iterating over observers and trying to call the endpoint. I wouldn't refactor now because:
- now we can have custom logs for each function
- in case of a refactoring, I would start from a higher level (if the proxy endpoint only forwards the response from the nodes, then have some common functions like `HandleGenericGetRequest(path string, shardID uint32) (data.GenericApiResponse, error)` . I would leave this for another PR

Reply:
Add a TODO and a task for this. It is super easy to refactor this.

Reply:
added todo